kartothek/serialization/_parquet.py

```diff
@@ -56,9 +76,18 @@ def _reset_dictionary_columns(table):
     2. Massive performance issue due to non-fast path implementation https://issues.apache.org/jira/browse/ARROW-5089
     """
     for i in range(table.num_columns):
-        if pa.types.is_dictionary(table[i].type):
-            new_col = table[i].cast(table[i].data.chunk(0).dictionary.type)
-            table = table.remove_column(i).add_column(i, new_col)
+        col = table[i]
+        if pa.types.is_dictionary(col.type):
+            new_type = col.data.chunk(0).dictionary.type
+            new_col = col.cast(new_type)
+            if ARROW_LARGER_EQ_0150:
+                field = table.field(i)
+                new_field = pa.field(
+                    field.name, new_type, field.nullable, field.metadata
+                )
+                table = table.remove_column(i).add_column(i, new_field, new_col)
+            else:
+                table = table.remove_column(i).add_column(i, new_col)
     return table
```

Comment:
I was rather thinking

```python
def _reset_dictionary_columns(table):
    if ARROW_LARGER_EQ_0150:
        for i in range(table.num_columns):
            if pa.types.is_dictionary(table[i].type):
                new_col = table[i].cast(table[i].data.chunk(0).dictionary.type)
                table = table.remove_column(i).add_column(i, new_col)
        col = table[i]
        if pa.types.is_dictionary(col.type):
            new_type = col.data.chunk(0).dictionary.type
            new_col = col.cast(new_type)
            table = table.remove_column(i).add_column(i, new_col)
    return table
```

Sorry if this wasn't clear

Reply:
I think you mean "you don't need to implement the arrow 0.15.0 compat for _reset_dictionary_columns since it isn't used under arrow 0.15.0". Your code example doesn't make much sense to me though.

Reply:
the current code is doing what you proposed I think. the function isn't called under arrow 0.15.0